podaac/subsetter/subset.py

```diff
@@ -1032,15 +1083,17 @@ def subset(file_to_subset, bbox, output_file, variables=None,  # pylint: disable
         subset to be performed. If this value is not provided, the
         granule will not be subset temporally on the upper bound.
     """
+    file_extension = file_to_subset.split('.')[-1]
+    if file_extension in ('nc4', 'nc'):
+        # Open dataset with netCDF4 first, so we can get group info
+        nc_dataset = nc.Dataset(file_to_subset, mode='r')
+        has_groups = bool(nc_dataset.groups)
 
-    # Open dataset with netCDF4 first, so we can get group info
-    nc_dataset = nc.Dataset(file_to_subset, mode='r')
-
-    has_groups = bool(nc_dataset.groups)
-
-    # If dataset has groups, transform to work with xarray
-    if has_groups:
-        nc_dataset = transform_grouped_dataset(nc_dataset, file_to_subset)
+        # If dataset has groups, transform to work with xarray
+        if has_groups:
+            nc_dataset = transform_grouped_dataset(nc_dataset, file_to_subset)
+    elif file_extension == 'he5':
+        nc_dataset = h5file_transform(file_to_subset)
```

Comment:
I'm wondering if it would make sense to do something like

```python
    if file_extension == 'he5':
        nc_dataset = h5file_transform(file_to_subset)
    else:
        # Open dataset with netCDF4 first, so we can get group info
        nc_dataset = nc.Dataset(file_to_subset, mode='r')
        has_groups = bool(nc_dataset.groups)

        # If dataset has groups, transform to work with xarray
        if has_groups:
            nc_dataset = transform_grouped_dataset(nc_dataset, file_to_subset)
```

The reason for that would be to avoid breaking existing functionality. For example I'm pretty sure the subsetter works as-is on `SMAP_JPL_L2B_SSS_CAP_V5` which has extension `h5`.

Reply:
I guess a reason why I went with an elif, just so users know which file types are getting processed, and we'll need a separate .hdf file type

Reply:
@nlenssen2013 This will break subsetting capabilities for our `SMAP_JPL_L2B_SSS_CAP_V5` collection though, because it isn't `nc`, `nc4`, or `he5` type. You could either add that type to the list of types like

```
if file_extension in ('nc4', 'nc', 'h5'):
```

or something like my original suggestion. I would probably lean towards my original suggestion because otherwise, we only support a hardcoded list of types which doesn't seem very flexible

Reply:
@skorper got it, updated now